### PR TITLE
feat(flags): Set up initial UI-reorganization for feature management

### PR DESCRIPTION
**Pull request by havenbarnes**
## Problem

Initial setup for: https://github.com/PostHog/posthog/issues/25938

Right now this is just a UI re-organization and does not have any functional changes on the sub-pages themselves. The next step is to customize the Experiments tab to only show relevant items, recall from RFC "The "Experiments" section will be subset of all the experiments we have that map to features"

## Changes

Screenshot 2024-12-02 at 5 22 27 PM

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

https://www.loom.com/share/a1ab3753e5374844ad2c162ac8b61d79

### CODE REVIEW

**Comment by raquelmsmith** [comment]
I think the left nav item should be "Features" and go at the top under Notebooks

I was also thinking that Features would be a list of features - eg "new friend inviter" or "document creator" - these are _not_ features we offer in posthog, but instead a thing that someone makes to track their own feature development and rollout. 

So each "feature" should have a section for "linked flags" which is the flags that someone has said are related to the feature. Same for experiments. 

We can chat through this on monday if you'd like!

File: frontend/src/scenes/feature-flags/FeatureManagementScenes.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { Spinner } from 'lib/lemon-ui/Spinner'
+import { lazy, Suspense } from 'react'
+
+const EarlyAccessFeatures = lazy(() =>
```
Comment (havenbarnes) [inline frontend/src/scenes/feature-flags/FeatureManagementScenes.tsx:4]: note for reviewing later: these lazy imports are working well in local dev, but it is unprecedented in the app. Check to make sure there isn't some reason for not using it before. If so, it's a superfluous optimization anyway, but nice for speedy loads

File: frontend/src/scenes/feature-flags/featureManagementLogic.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                setActiveFeatureId: (_, { activeFeatureId }) => activeFeatureId,
+            },
+        ],
+    }),
+    loaders(({ values }) => ({
+        features: [
+            { results: [], count: 0, offset: 0 } as FeaturesResult,
+            {
+                loadFeatures: async () => {
+                    const response = await api.get(`api/projects/${values.currentTeamId}/features`)
```
Comment (havenbarnes) [inline frontend/src/scenes/feature-flags/featureManagementLogic.ts:45]: I'll be swapping out this raw URL for a proper entry in `api.ts` in the next incoming PR

File: frontend/src/layout/navigation-3000/navigationLogic.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import {
     IconDashboard,
     IconDatabase,
     IconDecisionTree,
+    IconFeatures,
```
Comment (dmarticus) [inline frontend/src/layout/navigation-3000/navigationLogic.tsx:7]: [nit] not your fault at all, but it's kinda annoying how our feature icon and our AI icon are both iterations of the sparkle icon.  Feels like we could come up with something better here for our features _product_, but not sure if it's worth the effort yet.

**Review by dmarticus — APPROVED** [review]
Early days but seems reasonable to start; though not much crazy stuff here yet.  I'd say ship this behind the flag and keep on iterating.

File: frontend/src/layout/navigation-3000/navigationLogic.tsx (hunk 3 above)
Comment (havenbarnes) [reply frontend/src/layout/navigation-3000/navigationLogic.tsx:7]: Agree on both fronts; let's get something new but asset swapping can wait til real launch!

**Pull request merged**